tracer/src/Datadog.Trace.ClrProfiler.Managed.Loader/Startup.NetFramework.cs

```diff
@@ -60,25 +70,39 @@ private static Assembly ResolveAssembly(string name)
 
             // WARNING: Logs must not be added _before_ we check for the above bail-out conditions
             var path = string.IsNullOrEmpty(ManagedProfilerDirectory) ? $"{assemblyName.Name}.dll" : Path.Combine(ManagedProfilerDirectory, $"{assemblyName.Name}.dll");
-            StartupLogger.Debug("  Looking for: '{0}'", path);
+            StartupLogger.Debug("Assembly Resolve event received for: {0}. Looking for: {1}", name, path);
 
             if (File.Exists(path))
             {
-                if (name.StartsWith("Datadog.Trace, Version=") && name != AssemblyName)
+                if (name.StartsWith("Datadog.Trace, Version=", StringComparison.Ordinal) && name != AssemblyName)
                 {
                     StartupLogger.Debug("  Trying to load '{0}' which does not match the expected version ('{1}'). [Path={2}]", name, AssemblyName, path);
                     return null;
                 }
 
-                StartupLogger.Debug("  Resolving '{0}', loading '{1}'", name, path);
+                StartupLogger.Debug("Calling Assembly.LoadFrom(\"{0}\")", path);
                 var assembly = Assembly.LoadFrom(path);
-                StartupLogger.Debug("Assembly '{0}' loaded.", assembly?.FullName ?? "(null)");
+                StartupLogger.Debug("Assembly loaded: {0}", assembly.FullName);
                 return assembly;
             }
 
-            StartupLogger.Debug("Assembly not found in path: '{0}'", path);
+            StartupLogger.Debug("Assembly not found in path: {0}", path);
             return null;
         }
+
+        private static IEnumerable<string> EnumerateProfilerPathEnvironmentVariables()
```

Comment:
same argument, I'd rather inline this than have the itterator implementation personally

Reply:
refactored